### PR TITLE
feat: refactor with typescript

### LINKED ISSUE #116: need to refactor with typescript ?

### PR DESCRIPTION
**Pull request by liuhanqu**
BREAKING CHANGE: Drop Node.js < 16 support

closes https://github.com/node-modules/ready-callback/issues/116

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "extends": "@eggjs/tsconfig",
+  "compilerOptions": {
+    "target": "es2022",
+    "module": "nodenext",
+    "moduleResolution": "nodenext"
+  },
+  "include": [
```
Comment (fengmk2) [inline tsconfig.json]: no need to set include and exclude settings.

File: src/index.ts [hunk 2]
```diff
@@ -13,21 +32,19 @@ const defaults = {
...
-  constructor(opt) {
+  opt: ReadyOption;
+  obj: any;
+
+  ready: (flagOrFunction?: ReadyFunctionArg) => void;
+
+  constructor(opt: ReadyOption = {}) {
     super();
-    ready.mixin(this);
+    ReadyObject.default.mixin(this);
```
Comment (fengmk2) [inline src/index.ts]: https://github.com/node-modules/get-ready/pull/3 支持 esm 了。

File: tsconfig.json (hunk 1 above)
Comment (liuhanqu) [reply tsconfig.json]: have deleted

File: src/index.ts (hunk 2 above)
Comment (liuhanqu) [reply src/index.ts]: have updated

**Review by fengmk2 — APPROVED** [review]

**Comment by fengmk2** [comment]
@liuhanqu Nice job!

**Pull request merged**